### PR TITLE
[RBSetV2] Add ability to change entry fee

### PR DESCRIPTION
**Pull request by felix2feng**
- Refactors common assertions into single formulas
- Add ability to change entry fee
- Adds the following state variables: `auctionLibrary`, `proposalPeriod`, `proposalStartTime`

### CODE REVIEW

File: contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol [hunk 1]
```diff
@@ -187,6 +193,20 @@ contract RebalancingSetState {
         manager = _newManager;
     }
 
+    function setEntryFee(
```
Comment (bweick) [inline contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol]: Can you add this to IRebalancingSetTokenV2 as well

File: contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol [hunk 2]
```diff
@@ -127,10 +131,7 @@ contract RebalancingSetState {
     /* ============ Modifier ============ */
 
     modifier onlyManager() {
-        require(
-            msg.sender == manager,
-            "Must be manager"
-        );
+        validateManager();
```
Comment (bweick) [inline contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol]: Are we ever using this in a non-modifier way? Don't know why it needs to be a separate function

Comment (felix2feng) [reply contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol]: It's more bytecode-efficient to separate the function out, as each piece of logic and the reason strings are duplicated across each function that utilizes this modifier

File: contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol (hunk 1 above)
Comment (felix2feng) [reply contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol]: sure

File: contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol (hunk 2 above)
Comment (bweick) [reply contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol]: I see, ok that makes sense

File: contracts/core/fee-calculators/FixedFeeCalculator.sol [hunk 3]
```diff
@@ -36,8 +37,6 @@ contract FixedFeeCalculator is IFeeCalculator {
 
     /* ============ Constants ============ */
```
Comment (bweick) [inline contracts/core/fee-calculators/FixedFeeCalculator.sol]: Don't need a constants section anymore

File: contracts/lib/ScaleValidations.sol [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+pragma solidity 0.5.7;
+
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+
+library ScaleValidations {
+    using SafeMath for uint256;
+
+    uint256 private constant ONE_HUNDRED_PERCENT = 10 ** 18;
```
Comment (bweick) [inline contracts/lib/ScaleValidations.sol]: One thing I like the ABDK guys suggested was to start using `1e14`. Didn't realize that was a pattern you could use in Solidity.

File: contracts/lib/ScaleValidations.sol [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    uint256 private constant ONE_BASIS_POINT = 10 ** 14;
+
+    function validateLessThanEqualOneHundredPercent(uint256 _value) internal view {
+        require(_value <= ONE_HUNDRED_PERCENT, "Must be < 100%");
```
Comment (bweick) [inline contracts/lib/ScaleValidations.sol]: <= in msg

File: contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol [hunk 6]
```diff
@@ -17,6 +17,8 @@
 pragma solidity 0.5.7;
 pragma experimental "ABIEncoderV2";
 
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
```
Comment (bweick) [inline contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol]: Where is `SafeMath` being used?

**Pull request merged**